src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/events/handlers/MigrateVideoBookingEventHandler.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.handlers
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.client.whereaboutsapi.WhereaboutsApiClient
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.repository.VideoBookingRepository
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.TelemetryService
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.DomainEvent
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.DomainEventType
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.VideoBookingInformation
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.migration.MigrateVideoBookingService
+
+@Component
+class MigrateVideoBookingEventHandler(
+  private val videoBookingRepository: VideoBookingRepository,
+  private val whereaboutsApiClient: WhereaboutsApiClient,
+  private val migrateVideoBookingService: MigrateVideoBookingService,
+  private val telemetryService: TelemetryService,
+) :
+  DomainEventHandler<MigrateVideoBookingEvent> {
+  override fun handle(event: MigrateVideoBookingEvent) {
+    if (bookingToMigrateDoesNotAlreadyExist(event)) {
+      runCatching {
+        val bookingToMigrate =
+          whereaboutsApiClient.findBookingDetails(event.additionalInformation.videoBookingId)
+            ?: throw NullPointerException(
```

Comment:
Throwing a NPE here might not be appropriate as it will go back on the queue for a retry.

Technically it should never happen, we would only be searching for it on the back of a migration event from whereabouts-api.